server/functions/0_parameters.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package functions
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+// ParameterType represents the type of a parameter.
+type ParameterType uint8
+
+const (
+	ParameterType_Null      ParameterType = iota // The parameter is a NULL value, and is therefore typeless
+	ParameterType_Integer                        // The parameter is an Integer type
+	ParameterType_Float                          // The parameter is a Float type
+	ParameterType_Numeric                        // The parameter is a Numeric type
+	ParameterType_String                         // The parameter is a String type
+	ParameterType_Timestamp                      // The parameter is a Timestamp type
+
+	ParameterType_Length // The number of parameters. This should always be last in the enum declaration.
+)
+
+// ptCasts contains an array of all potential casts for each parameter type
+var ptCasts = [ParameterType_Length][]ParameterType{
```

Comment:
This would be more clearly expressed with an init block that assigns each index with the constants above